packages/frontend-plugin-api/src/wiring/resolveExtensionDefinition.ts

```diff
@@ -114,15 +114,12 @@ export type ResolveExtensionId<
   TDefaultNamespace extends string | undefined,
 > = TExtension extends ExtensionDefinition<{
   kind: infer IKind extends string | undefined;
-  namespace: infer INamespace extends string | undefined;
   name: infer IName extends string | undefined;
 }>
-  ? [string | undefined] extends [IKind | INamespace | IName]
+  ? [string | undefined] extends [IKind | TDefaultNamespace | IName]
     ? never
     : (
-        (
-          undefined extends TDefaultNamespace ? INamespace : TDefaultNamespace
-        ) extends infer ISelectedNamespace extends string
+        TDefaultNamespace extends infer ISelectedNamespace extends string
           ? undefined extends IName
```

Comment:
No need for much of this anymore, can just make the param `TNamespace` and get rid of `ISelectedNamespace`